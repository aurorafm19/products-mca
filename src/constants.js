export const NOT_INFO_AVAILABLE = 'not info available';
export const NOT_SEARCH_INFO = 'Sorry, there are no results for your search!';
export const PRODUCT_DETAIL_BREADCRUMB = 'product detail';
export const HOUR_EXPIRATION = 3600000;
export const SEARCHBAR_PLACEHOLDER = 'search by brand or model';
