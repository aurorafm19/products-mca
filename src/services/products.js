const apiUrl = 'https://front-test-api.herokuapp.com/api/';
const headers = { 'Content-Type': 'application/json'};

export function getProductlist() {
    return fetch(`${apiUrl}product`, {method: 'GET', headers});
}

export function getProductDetail(id) {
    return fetch(`${apiUrl}product/${id}`, {method: 'GET', headers})
    .then((response) => response.json());
}

export function addProduct(product) {
    fetch(`${apiUrl}/cart`, {method: 'POST', headers, body: JSON.stringify(product)} )
}

export function filterBySearchValue(searchValue, productList) {
    let productsFiltered = [];
    if (searchValue.length > 0) {
        productsFiltered = productList.filter(product =>product.brand === searchValue || product.model === searchValue )
    }

    return productsFiltered;
}
