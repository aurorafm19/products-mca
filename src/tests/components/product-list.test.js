import { render, fireEvent } from '@testing-library/react';
import ProductList from '../../components/product-list';
import {SEARCHBAR_PLACEHOLDER} from '../../constants';
import {PRODUCTS} from '../mocks/products';

const build = () => {
    const {container, getByPlaceholderText} = render(<ProductList />);

    return {
        container,
        getByPlaceholderText
    };
}
test('renders the component with no results', () => {
    const {container} = build();

    expect(container).toMatchSnapshot();
});
 