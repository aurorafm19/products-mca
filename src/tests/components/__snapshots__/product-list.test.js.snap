// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component with no results 1`] = `
<div>
  <div
    class="product-list"
  >
    <div
      class="searchbar"
    >
      <svg
        class="search-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <input
        class="searchbar-input"
        placeholder="search by brand or model"
        type="text"
        value=""
      />
      <svg
        class="close-icon"
        data-testid="close-button"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
    <div
      class="products-card-container not-found-products"
    >
      <svg
        class="not-found-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="15.5"
          cy="9.5"
          r="1.5"
        />
        <circle
          cx="8.5"
          cy="9.5"
          r="1.5"
        />
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-6c-2.33 0-4.32 1.45-5.12 3.5h1.67c.69-1.19 1.97-2 3.45-2s2.75.81 3.45 2h1.67c-.8-2.05-2.79-3.5-5.12-3.5z"
        />
      </svg>
      Sorry, there are no results for your search!
    </div>
  </div>
</div>
`;
